ios/CodePush/CodePush.m

```diff
@@ -208,45 +208,64 @@ + (void)ensureBinaryBundleExists
 {
     if (![self binaryBundleURL]) {
         NSString *errorMessage;
-        
+
 #if TARGET_IPHONE_SIMULATOR
         errorMessage = @"React Native doesn't generate your app's JS bundle by default when deploying to the simulator. "
         "If you'd like to test CodePush using the simulator, you can do one of three things depending on your React "
         "Native version and/or preferred workflow:\n\n"
-        
+
         "1. Update your AppDelegate.m file to load the JS bundle from the packager instead of from CodePush. "
         "You can still test your CodePush update experience using this workflow (debug builds only).\n\n"
-        
+
         "2. Force the JS bundle to be generated in simulator builds by removing the if block that echoes "
         "\"Skipping bundling for Simulator platform\" in the \"node_modules/react-native/packager/react-native-xcode.sh\" file.\n\n"
-        
+
         "3. Deploy a release build to the simulator, which unlike debug builds, will generate the JS bundle (React Native >=0.22.0 only).";
 #else
         errorMessage = [NSString stringWithFormat:@"The specified JS bundle file wasn't found within the app's binary. Is \"%@\" the correct file name?", [bundleResourceName stringByAppendingPathExtension:bundleResourceExtension]];
 #endif
-        
+
         RCTFatal([CodePushErrorUtils errorWithMessage:errorMessage]);
     }
 }
 
 - (instancetype)init
 {
     self = [super init];
-    
+
     if (self) {
         [self initializeUpdateAfterRestart];
     }
-    
+
     return self;
 }
 
+- (void)clearDebugUpdates
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if ([_bridge.bundleURL.scheme hasPrefix:@"http"]) {
+            NSError *error;
+            NSString *binaryAppVersion = [[CodePushConfig current] appVersion];
+            NSDictionary *currentPackageMetadata = [CodePushPackage getCurrentPackage:&error];
+            if (currentPackageMetadata) {
+                NSString *packageAppVersion = [currentPackageMetadata objectForKey:AppVersionKey];
+                if (![binaryAppVersion isEqualToString:packageAppVersion]) {
+                    [CodePush clearUpdates];
+                }
+            }
+        }
+    });
+}
+
 /*
  * This method is used when the app is started to either
  * initialize a pending update or rollback a faulty update
  * to the previous version.
  */
 - (void)initializeUpdateAfterRestart
 {
+    [self clearDebugUpdates];
```

Comment:
Should we actually wrap this in an #ifdef DEBUG statement? That way it's limited to just debug builds? I can't imagine why someone would create a release build that is using the packager, and I'm in favor of optimizing release builds as much as possible.